### PR TITLE
fix(2fa): avoid caching bad 2fa token

### LINKED ISSUE #1309: Expired OTP is inadvertently cached
When an OTP expires, octokit generates a new one via `on2fa` but doesn't use it.

The ordering of arguments passed to `Object.assign` at https://github.com/octokit/rest.js/blob/eb1ebab975cb3a5877e7df25acc500b3e048b49e/plugins/authentication/request-error.js#L32 means that even though it calls the `on2fa()` function and generates a new OTP, the otp already in `options.headers` is the one that ultimately gets used.

Reversing the order of arguments fixes the problem for me. I'm happy to open a PR if that's the preferred fix, but I'm not very familiar with the codebase and I wonder if it's more appropriate to either invalidate something in `state` or delete the header.

### PR DESCRIPTION
**Pull request by ianwremmel**
closes #1309

### CODE REVIEW

File: test/integration/authentication-test.js [hunk 1]
```diff
@@ -155,6 +155,66 @@ describe('authentication', () => {
...
+
+    let callCount = 0
+    const octokit = new Octokit({
+      baseUrl: 'https://authentication-test-host.com',
+      auth: {
+        username: 'username',
+        password: 'password',
+        on2fa () {
+          callCount++
+          return callCount
```
Comment (gr2m) [inline test/integration/authentication-test.js]: you can replace the two lines above with

```js
return ++callCount
```

**Review by gr2m — CHANGES_REQUESTED** [review]
This looks great, just a tiny code simplification. I can fix that up, too, just let me know :)

File: test/integration/authentication-test.js (hunk 1 above)
Comment (ianwremmel) [reply test/integration/authentication-test.js]: changed

**Review by gr2m — APPROVED** [review]
Thanks a lot Ian 💝

**Comment by gr2m** [comment]
Also thanks @ruxandrafed for your review 👍

**Pull request merged**